pulpcore/pulpcore/app/viewsets/custom_filters.py

```diff
@@ -10,3 +13,31 @@ class CharInFilter(filters.BaseInFilter, filters.CharFilter):
     than one object in a single query.
     """
     pass
+
+
+class HyperlinkRelatedFilter(filters.Filter):
+    """
+    Enables a user to filter by a foreign key using that FK's href
+    """
+
+    def filter(self, qs, value):
+        """
+        Args:
+            qs (django.db.models.query.QuerySet): The Queryset to filter
+            value (string): href representing an instance of foreign key
+
+        Returns:
+            Queryset filtered by the foreign key instance
+        """
+
+        if not value:
+            return qs
+
+        try:
+            match = resolve(urlparse(value).path)
+        except Exception:
+            return qs
```

Comment:
I'm a bit worried this might, in case one (by mistake) specifies a malformed URL, not limit the returned list of objects (tasks): `GET@/v3/tasks/?worker=this_aint_url&state__in=waiting` might return more tasks than expected. Shouldn't in such a case a 400 response be generated instead? Or a 404?